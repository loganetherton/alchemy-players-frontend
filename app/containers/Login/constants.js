export const CHANGE_EMAIL = 'playerApp/Login/CHANGE_EMAIL';
export const CHANGE_PASSWORD = 'playerApp/Login/CHANGE_PASSWORD';
export const LOGIN = 'playerApp/Login/LOGIN';
export const LOGIN_SUCCESS = 'playerApp/Login/LOGIN_SUCCESS';
export const LOGIN_FAILED = 'playerApp/Login/LOGIN_FAILED';
export const RESET_LOGIN = 'playerApp/Login/RESET_LOGIN';
