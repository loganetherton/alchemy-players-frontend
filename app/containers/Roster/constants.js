export const GET_PLAYERS = 'playerApp/Players/GET_PLAYERS';
export const GET_PLAYERS_SUCCESS = 'playerApp/Players/GET_PLAYERS_SUCCESS';
export const GET_PLAYERS_FAILURE = 'playerApp/Players/GET_PLAYERS_FAILURE';
export const CHANGE_FIRST_NAME = 'playerApp/Players/CHANGE_FIRST_NAME';
export const CHANGE_LAST_NAME = 'playerApp/Players/CHANGE_LAST_NAME';
export const CHANGE_RATING = 'playerApp/Players/CHANGE_RATING';
export const CHANGE_HANDEDNESS = 'playerApp/Players/CHANGE_HANDEDNESS';
export const CREATE_PLAYER = 'playerApp/Players/CREATE_PLAYER';
export const CREATE_PLAYER_SUCCESS = 'playerApp/Players/CREATE_PLAYER_SUCCESS';
export const CREATE_PLAYER_FAILURE = 'playerApp/Players/CREATE_PLAYER_FAILURE';
export const DELETE_PLAYER = 'playerApp/Players/DELETE_PLAYER';
export const DELETE_PLAYER_SUCCESS = 'playerApp/Players/DELETE_PLAYER_SUCCESS';
export const DELETE_PLAYER_FAILURE = 'playerApp/Players/DELETE_PLAYER_FAILURE';
export const RESET_PLAYER_FORM = 'playerApp/Players/RESET_PLAYER_FORM';
