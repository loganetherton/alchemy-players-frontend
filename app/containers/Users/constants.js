/*
 * HomeConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const CHANGE_FIRST_NAME = 'playerApp/Login/CHANGE_FIRST_NAME';
export const CHANGE_LAST_NAME = 'playerApp/Login/CHANGE_LAST_NAME';
export const CHANGE_EMAIL = 'playerApp/Login/CHANGE_EMAIL';
export const CHANGE_PASSWORD = 'playerApp/Login/CHANGE_PASSWORD';
export const CHANGE_CONFIRM_PASSWORD = 'playerApp/Login/CHANGE_CONFIRM_PASSWORD';
export const CREATE_USER = 'playerApp/Login/CREATE_USER';
export const USER_CREATED = 'playerApp/Login/USER_CREATED';
export const USER_CREATION_FAILED = 'playerApp/Login/USER_CREATION_FAILED';
export const RESET_USER_CREATION = 'playerApp/Login/RESET_USER_CREATION';
