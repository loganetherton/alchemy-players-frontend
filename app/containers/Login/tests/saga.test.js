/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { CHANGE_EMAIL, CHANGE_PASSWORD, LOGIN_SUCCESS, LOGIN_FAILED, RESET_LOGIN, LOGIN} from '../constants';
import {
  changeEmail,
  changePassword,
  loginSuccess,
} from '../actions';

import login, {doLogin} from '../saga';

const username = 'mxstbr';

/* eslint-disable redux-saga/yield-effects */
describe('Login Saga', () => {
  let loginGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    loginGenerator = doLogin();

    const selectDescriptor = loginGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = loginGenerator.next(username).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the reposLoaded action if it requests the data successfully', () => {
    const response = {"success":true,"token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1YjhkZDBhMTMxMGQyMTdmZjM5MDE4ODQiLCJpYXQiOjE1MzY2MDU5Mjh9.oZEfwR29UnQwPEgWH1b3Phr30VHROWAZs8RQteLnY_A","user":{"first_name":"blahblah12","last_name":"blahblah12","email":"blahblah12@blahblah12.com","id":"5b8dd0a1310d217ff3901884"}};
    const putDescriptor = loginGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(loginSuccess(response)));
  });
});
