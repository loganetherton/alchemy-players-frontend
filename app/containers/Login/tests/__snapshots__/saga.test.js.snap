// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Saga should dispatch the reposLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`Login Saga should dispatch the reposLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://players-api.developer.alchemy.codes/api/login",
      Object {
        "body": "mxstbr",
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
